json export is failing if the directory log_parser_output/exports is not existing

### Expected Behaviour
By calling the methods exportLogDataToJSON, if the log_parser_output/exports directory is not existing, we expect to have this one created by the log-parser.

It's the case for HTML and csv export, and it's working.

### Actual Behaviour
Currently, if the job calling the lob parser didn't create explicitly the directory log_parser_output/exports, it's failing.

The log parser directory for extraction (log_parser_output/exports) is defined here:
https://github.com/adobe/log-parser/blob/a84ea801b82ad0aed7a3ec346ac8245377f25018/src/main/java/com/adobe/campaign/tests/logparser/utils/LogParserFileUtils.java#L18

In the case of Json export, if the directory is not explicity created before, it's failing with the issue:
```
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:3.4.1:java (default-cli) 
on project parseCampaignLogs: An exception occurred while executing the Java class. 
Encountered error while exporting the log data to a JSON file.: 
log_parser_output/exports/logsExtraction.json (No such file or directory) -> [Help 1]
```

The stack trace is:

```
com.adobe.campaign.tests.logparser.exceptions.LogDataExportToFileException: Encountered error while exporting the log data to a JSON file.
    at com.adobe.campaign.tests.logparser.core.LogData.exportLogDataToJSON (LogData.java:548)
    at com.adobe.campaign.tests.logparser.core.LogData.exportLogDataToJSON (LogData.java:521)
    at com.adobe.campaign.tests.logs.ExtractACCLogs.main (ExtractACCLogs.java:74)
    at org.codehaus.mojo.exec.ExecJavaMojo.doMain (ExecJavaMojo.java:358)
    at org.codehaus.mojo.exec.ExecJavaMojo.doExec (ExecJavaMojo.java:347)
    at org.codehaus.mojo.exec.ExecJavaMojo.lambda$execute$0 (ExecJavaMojo.java:269)
    at java.lang.Thread.run (Thread.java:834)
Caused by: java.io.FileNotFoundException: log_parser_output/exports/logsExtraction.json (No such file or directory)
```

--- a/README.md
+++ b/README.md
@@ -71,7 +71,7 @@ The following dependency needs to be added to your pom file:
  <dependency>
     <groupId>com.adobe.campaign.tests</groupId>
     <artifactId>log-parser</artifactId>
-    <version>1.11.0</version>
+    <version>1.11.1</version>
 </dependency>
 ```
 ## Running the Log Parser
@@ -470,6 +470,9 @@ You can get a print out of the command line options by running the command with 
 All reports are stored in the directory `log-parser-reports/export/`.
 
 ## Changelog
+### 1.11.1
+- [#188](https://github.com/adobe/log-parser/issues/188) We solved problems with exporting when the directory hierarchy is not present.
+
 ### 1.11.0
 - **(new feature)** [#10](https://github.com/adobe/log-parser/issues/10) We now have an executable for the log-parser. You can perform a log parsing using the command line. For more information please read the section on [Command-line Execution of the Log-Parser](#command-line-execution-of-the-log-parser).
 - **(new feature)** [#127](https://github.com/adobe/log-parser/issues/127) You can now compare two LogData Objects. This is a light compare that checks that for a given key, if it is absent, added or changes in frequency.

--- a/src/main/java/com/adobe/campaign/tests/logparser/core/LogData.java
+++ b/src/main/java/com/adobe/campaign/tests/logparser/core/LogData.java
@@ -415,19 +415,24 @@ public class LogData<T extends StdLogEntry> {
      */
     public File exportLogDataToCSV(Collection<String> in_headerSet, String in_csvFileName)
             throws LogDataExportToFileException {
-        File l_exportFile = LogParserFileUtils.createNewFile(in_csvFileName);
+        File l_exportFile = null;
+        try {
+            l_exportFile = LogParserFileUtils.createNewFile(in_csvFileName);
 
-        try (CSVPrinter printer = new CSVPrinter(new FileWriter(l_exportFile), CSVFormat.DEFAULT)) {
-            printer.printRecord(in_headerSet);
+            try (CSVPrinter printer = new CSVPrinter(new FileWriter(l_exportFile), CSVFormat.DEFAULT)) {
+                printer.printRecord(in_headerSet);
 
-            for (StdLogEntry lt_entry : this.getEntries().values()) {
-                Map lt_values = lt_entry.fetchValueMapPrintable();
-                printer.printRecord(in_headerSet.stream().map(h -> lt_values.get(h)).collect(Collectors.toList()));
+                for (StdLogEntry lt_entry : this.getEntries().values()) {
+                    Map lt_values = lt_entry.fetchValueMapPrintable();
+                    printer.printRecord(in_headerSet.stream().map(h -> lt_values.get(h)).collect(Collectors.toList()));
+                }
             }
 
         } catch (IOException ex) {
             throw new LogDataExportToFileException("Encountered error while exporting the log data to a CSV file.", ex);
         }
+
+
 
         return l_exportFile;
     }
@@ -459,9 +464,10 @@ public class LogData<T extends StdLogEntry> {
      * @return an HTML file containing the LogData as a table
      */
     public File exportLogDataToHTML(Collection<String> in_headerSet, String in_reportTitle, String in_htmlFileName) {
-        File l_exportFile = LogParserFileUtils.createNewFile(in_htmlFileName);
+        File l_exportFile;
 
         try {
+            l_exportFile = LogParserFileUtils.createNewFile(in_htmlFileName);
             StringBuilder sb = new StringBuilder();
             sb.append(HTMLReportUtils.fetchSTDPageStart("diffTable.css"));
             //Creating the overview report
@@ -485,7 +491,7 @@ public class LogData<T extends StdLogEntry> {
 
             FileUtils.writeStringToFile(l_exportFile, sb.toString(), "UTF-8");
         } catch (IOException e) {
-            throw new LogDataExportToFileException("We were unable to write to the file " + l_exportFile.getPath());
+            throw new LogDataExportToFileException("We were unable to write to the file " + in_htmlFileName, e);
         }
 
         return l_exportFile;
@@ -535,15 +541,18 @@ public class LogData<T extends StdLogEntry> {
      */
     public File exportLogDataToJSON(Collection<String> in_headerSet, String in_jsonFileName)
             throws LogDataExportToFileException {
-        File l_exportFile = LogParserFileUtils.createNewFile(in_jsonFileName);
-        List<Map<String, String>> jsonList = new ArrayList<>();
-        jsonList.addAll(this.getEntries().values().stream().map(StdLogEntry::fetchValueMapPrintable)
-                .collect(Collectors.toList()));
+        File l_exportFile;
 
         try {
-            ObjectMapper objectMapper = new ObjectMapper();
-            objectMapper.writeValue(l_exportFile,
-                    objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(jsonList));
+            l_exportFile= LogParserFileUtils.createNewFile(in_jsonFileName);
+            List<Map<String, String>> jsonList = new ArrayList<>();
+            jsonList.addAll(this.getEntries().values().stream().map(StdLogEntry::fetchValueMapPrintable)
+                    .collect(Collectors.toList()));
+
+
+                ObjectMapper objectMapper = new ObjectMapper();
+                objectMapper.writeValue(l_exportFile,
+                        objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(jsonList));
         } catch (IOException e) {
             throw new LogDataExportToFileException("Encountered error while exporting the log data to a JSON file.", e);
         }

--- a/src/main/java/com/adobe/campaign/tests/logparser/core/LogDataFactory.java
+++ b/src/main/java/com/adobe/campaign/tests/logparser/core/LogDataFactory.java
@@ -267,9 +267,10 @@ public class LogDataFactory {
             List<String> in_headers, String in_reportName) {
         StringBuilder sb = new StringBuilder();
         File l_exportFile = null;
-
+        String l_exportFilePath = in_reportName + ".html";
         try {
-            l_exportFile = LogParserFileUtils.createNewFile(in_reportName + ".html");
+            l_exportFile = LogParserFileUtils.createNewFile(l_exportFilePath);
+            l_exportFilePath = l_exportFile.getPath();
             sb.append(HTMLReportUtils.fetchSTDPageStart("diffTable.css"));
 
             //Creating the overview report
@@ -320,8 +321,9 @@ public class LogDataFactory {
             sb.append("</html>");
 
             FileUtils.writeStringToFile(l_exportFile, sb.toString(), "UTF-8");
+
         } catch (IOException e) {
-            throw new LogDataExportToFileException("We were unable to write to the file "+ l_exportFile.getPath());
+            throw new LogDataExportToFileException("We were unable to write to the file "+ l_exportFilePath);
         }
         return l_exportFile;
     }

--- a/src/main/java/com/adobe/campaign/tests/logparser/core/LogDataFactory.java
+++ b/src/main/java/com/adobe/campaign/tests/logparser/core/LogDataFactory.java
@@ -266,9 +266,10 @@ public class LogDataFactory {
     public static <T extends StdLogEntry> File generateDiffReport(Map<String, LogDataComparison<T>> in_comparisonReport,
             List<String> in_headers, String in_reportName) {
         StringBuilder sb = new StringBuilder();
-        File l_exportFile = LogParserFileUtils.createNewFile(in_reportName + ".html");
+        File l_exportFile = null;
 
         try {
+            l_exportFile = LogParserFileUtils.createNewFile(in_reportName + ".html");
             sb.append(HTMLReportUtils.fetchSTDPageStart("diffTable.css"));
 
             //Creating the overview report

--- a/src/main/java/com/adobe/campaign/tests/logparser/utils/LogParserFileUtils.java
+++ b/src/main/java/com/adobe/campaign/tests/logparser/utils/LogParserFileUtils.java
@@ -13,6 +13,9 @@ import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
 
 public class LogParserFileUtils {
     public static final String LOG_PARSER_EXPORTS = "log_parser_output/exports";
@@ -34,8 +37,8 @@ public class LogParserFileUtils {
      * @param in_fileName a file name
      * @return the file object
      */
-    public static File createFile(String in_fileName) {
-
+    public static File createFile(String in_fileName) throws IOException {
+        Files.createDirectories(Paths.get(LOG_PARSER_EXPORTS));
         return new File(LOG_PARSER_EXPORTS,in_fileName);
     }
 
@@ -44,7 +47,7 @@ public class LogParserFileUtils {
      * @param in_fileName A file name
      * @return a file object which is completely new and empty
      */
-    public static File createNewFile(String in_fileName) {
+    public static File createNewFile(String in_fileName) throws IOException {
         File lr_file = createFile(in_fileName);
         cleanFile(lr_file);
         return lr_file;

--- a/src/test/java/com/adobe/campaign/tests/logparser/core/LogDataTest.java
+++ b/src/test/java/com/adobe/campaign/tests/logparser/core/LogDataTest.java
@@ -21,11 +21,11 @@ import org.apache.commons.io.filefilter.WildcardFileFilter;
 import org.hamcrest.Matcher;
 import org.hamcrest.Matchers;
 import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
-import java.io.BufferedReader;
 import java.io.File;
-import java.io.FileReader;
 import java.io.IOException;
 import java.util.*;
 
@@ -34,6 +34,18 @@ import static org.hamcrest.Matchers.*;
 import static org.testng.Assert.assertThrows;
 
 public class LogDataTest {
+
+    @BeforeMethod(alwaysRun = true)
+    @AfterClass
+    public void CleanUp() throws IOException {
+        File l_stdOutPutFile = new File(LogParser.OUTPUT_ROOT);
+
+        if (l_stdOutPutFile.exists()) {
+            FileUtils.deleteDirectory(l_stdOutPutFile);
+        }
+
+    }
+
 
     /**
      * Testing that we correctly create a cube
@@ -1397,7 +1409,9 @@ public class LogDataTest {
 
         //Create the file so it is deleted
         File l_duplicateFile = new File(LogParserFileUtils.LOG_PARSER_EXPORTS, l_fileNameToExpect);
-        l_duplicateFile.createNewFile();
+        //Files.
+        //l_duplicateFile.createNewFile();
+        LogParserFileUtils.createNewFile(LogParserFileUtils.LOG_PARSER_EXPORTS+"/"+l_fileNameToExpect);
 
         assertThat("We should have the key for amcDataSource",
                 l_logData.getEntries().containsKey(l_searchItem1));

